spec/models/reservation_spec.rb

```diff
@@ -21,6 +21,47 @@
   # can't run if nil (?)
   #
 
+  describe '.number_for' do
+    before(:all) do
+      @ems = FactoryGirl.create_pair(:equipment_model)
+    end
+    before(:each) do
+      @source = FactoryGirl.build_pair(:valid_reservation,
+                                       equipment_model: @ems.first)
+    end
+    it 'counts the number that overlap with today' do
+      expect(Reservation.number_for(@source)).to eq 2
+    end
+    it 'counts the number that overlap with a given day' do
+      @source.last.start_date = Time.zone.today + 2
+      @source.last.due_date = Time.zone.today + 3
+      expect(Reservation.number_for(@source, date: Time.zone.today + 2.days)
+            ).to eq 1
+    end
+    it 'can count only not overdue reservations' do
+      @source.last.attributes = FactoryGirl.attributes_for(
+        :overdue_reservation)
+      @source.first.start_date = @source.last.start_date
+      expect(Reservation.number_for(@source, date: @source.last.start_date + 1,
+                                             overdue: false)).to eq 1
+    end
+    it 'can count only reservations on a specific model' do
+      @source.first.attributes = { equipment_model: @ems.last }
+      expect(Reservation.number_for(@source, equipment_model_id: @ems.last.id)
+            ).to eq 1
+    end
+    it 'can count not overdue reservations on a model' do
+      @source.last.attributes = FactoryGirl.attributes_for(
```

Comment:
Also, it doesn't match the equipment model attribute in any case, unless I'm missing something.